### PR TITLE
Rework "Backend is not available" error page

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
Rework the "Backend is not available" error page.

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/22511 and https://github.com/eclipse/che/issues/22484

### Is it tested? How?

1. Deploy Eclipse CHE.
-----
2. Create a new workspace and open the list of workspaces(`${che-server}/dashboard/#/workspaces`).
3. Open the dashboard in a new browser tab and press logout.
4. Return to the first tab and try to delete existing workspaces. It redirects to the login page.
----
5. Relogin and open the **Preferences** page and select **Container Registries**.
6. Use a patch ` oc patch deployment/che  -n eclipse-che --patch "{\"spec\":{\"replicas\":0}}"` to rolling out Che deployment.
7. Change selection from **Container Registries** to **Gitconfig**.
8. The following error message will appear: 'HTTP Error code 502. Endpoint which throws an error {che-server}/api/kubernetes/namespace/provision'. Check the browser logs message.'

Знімок екрана 2023-10-05 о 03 58 58

### CODE REVIEW

**Comment by ibuziuk** [comment]
@olexii4 we should not backport it for 3.9, this change should be only for 3.10

**Comment by ibuziuk** [comment]
@ScrewTSW could you please review the PR? The Happy path test seems to be failing again, restarted the job

**Review by ibuziuk — APPROVED** [review]

**Comment by ScrewTSW** [comment]
I got redirected successfully and the error message was shown
Screenshot_20231005_133554
Screenshot_20231005_133754
Screenshot_20231005_133816

**Review by ScrewTSW — APPROVED** [review]

File: packages/common/src/helpers/errors.ts [hunk 1]
```diff
@@ -50,7 +50,9 @@ export function getMessage(error: unknown): string {
   if (includesAxiosResponse(error)) {
     const response = error.response;
     if (typeof response.data === 'string') {
-      return response.data;
+      if (response.data.toLowerCase().trim().indexOf('<!doctype') !== 0) {
```
[the remark this answers is not in the document]
Comment (akurinnoy) [reply packages/common/src/helpers/errors.ts:53]: outdated diff

File: packages/dashboard-frontend/src/Layout/index.tsx [hunk 2]
```diff
@@ -135,7 +145,7 @@ export class Layout extends React.PureComponent<Props, State> {
           }
           isManagedSidebar={IS_MANAGED_SIDEBAR}
         >
-          <ErrorBoundary>
+          <ErrorBoundary ref={this.errorBoundaryRef}>
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/Layout/index.tsx]: I have fixed it.

File: packages/dashboard-frontend/package.json [hunk 3]
```diff
@@ -113,8 +112,8 @@
     "hard-source-webpack-plugin": "^0.13.1",
     "html-webpack-plugin": "^5.5.3",
     "identity-obj-proxy": "^3.0.0",
-    "jest": "^29.6.2",
-    "jest-environment-jsdom": "^29.6.2",
+    "jest": "^29.7.0",
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/package.json:115]: I have fixed it.

File: packages/dashboard-frontend/src/services/workspace-client/helpers.ts [hunk 4]
```diff
@@ -14,52 +14,92 @@ import common from '@eclipse-che/common';
...
+ */
+export function getErrorMessage(error: unknown): string {
+  let errorMessage = 'Check the browser logs message.';
+
+  if (typeof error === 'object' && error !== null) {
+    const { request, response } = error as { [propName: string]: any };
+    const code = response?.status ? response?.status : response?.request?.status;
+    const endpoint = request?.responseURL ? request?.responseURL : response?.request?.responseURL;
+
+    errorMessage = `HTTP Error code ${code}. Endpoint which throw an error ${endpoint}. ${errorMessage}`;
```
[the remark this answers is not in the document]
Comment (olexii4) [reply packages/dashboard-frontend/src/services/workspace-client/helpers.ts]: I have fixed it.

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**